content/configuration/communication/vault-csi.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Tokens from Vault via CSI driver"
+---
+
+This tutorial shows how to install BotKube which takes the configuration from Vault instance.
+
+## Prerequisites
+
+- Kubernetes Clusters that supports CSI.
+
+    For example, to run [K3s](https://k3s.io/) using [Lima](https://github.com/lima-vm/lima), run:
+    ```bash
+    limactl start template://k3s
+    ```
+
+- [Slack integration installed](http://localhost:1313/installation/slack/#install-botkube-slack-app-to-your-slack-workspace).
+- [`helm`](https://helm.sh/docs/intro/install/) v3 installed.
+- [`kubectl`](https://kubernetes.io/docs/tasks/tools/#kubectl) installed.
+
+
+## Steps
+
+This instruction guides you through the installation of BotKube and Vault on a Kubernetes cluster and configuring them together.
+
+1. Install Vault with CSI enabled:
+    ```bash
+    helm repo add hashicorp https://helm.releases.hashicorp.com
+    helm repo update
+    helm install vault hashicorp/vault --namespace default \
+        --set "server.dev.enabled=true" \
+        --set "injector.enabled=false" \
+        --set "csi.enabled=true"
+    ```
+
+2. Add Slack token in Vault:
+    ```bash
+    # Exec to pod
+    kubectl exec -n default -it vault-0 -- /bin/sh
+    ```
+    ```bash
+    # Write the token to Vault
+    vault kv put secret/slack token={token}
+    ```
+
+3. Enable Vault's Kubernetes authentication:
+    ```bash
+    vault auth enable kubernetes
+    vault write auth/kubernetes/config \
+        kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443"
+    ```
+    ```bash
+    vault policy write internal-app - <<EOF
+    path "secret/data/slack" {
+      capabilities = ["read"]
+    }
+    EOF
+    ```
+    ```bash
+    vault write auth/kubernetes/role/database \
+        bound_service_account_names=botkube-sa \
+        bound_service_account_namespaces=default \
+        policies=internal-app \
+        ttl=20m
+    ```
+
+4. Install the Secrets Store CSI driver:
+    ```bash
+    helm repo add secrets-store-csi-driver https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts
+    helm install csi secrets-store-csi-driver/secrets-store-csi-driver \
+    --namespace default \
+    --set syncSecret.enabled=true
+    ```
+
+5. Exit from the Vault Pod:
+    ```bash
+    exit
+    ```
```

Comment:
That should be done before point no 4. Actually, I would put it directly as the final command from point no 3.